package.json

```diff
@@ -39,18 +38,18 @@
     "semver": "^5.3.0"
   },
   "devDependencies": {
-    "@types/chai": "^4.0.1",
-    "@types/mocha": "^2.2.41",
-    "catch-uncommitted": "^1.0.0",
-    "chai": "^4.1.0",
-    "husky": "^0.14.3",
-    "jsdoc-to-markdown": "^3.0.0",
-    "lint-staged": "^4.0.4",
-    "mocha": "^3.4.2",
-    "prettier": "^1.6.1",
-    "ts-node": "^3.2.1",
-    "tslint": "^5.5.0",
-    "typescript": "^2.4.2"
+    "@types/chai": "^4.1.7",
+    "@types/mocha": "^5.2.6",
+    "catch-uncommitted": "^1.3.0",
+    "chai": "^4.2.0",
+    "husky": "^2.3.0",
+    "jsdoc-to-markdown": "^5.0.0",
+    "lint-staged": "^8.1.7",
+    "mocha": "^6.1.4",
+    "prettier": "^1.17.1",
+    "ts-node": "^7.0.1",
+    "tslint": "^5.16.0",
+    "typescript": "^3.4.5"
```

Comment:
I'm usually afraid that a TS bump could produce typings incompatible with the TS version of pre-existing consumers...
If we add this as a patch change and later for example add one of the newer TS keywords in our source (eg unknown/readonly) and that reaches the generated typings, then this would break new installations of the SDK & the CLI.

We could probably add a `"typesVersions": { ">=3.4": /*...*/ }` and point `"types"` to a v2.4.2 compatible definition file, but are there enough benefits for the effort that this requires?
See: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html

How about keeping this as it was for now?

Reply:
@thgreasi `typesVersions` is only used if you're running typescript >=3.1 so it won't help at all in this case where the previous version was <3.1, beyond that this package is in the far worse situation where it points directly at the typescript for typings instead of generating a declaration (I'll fix in another PR).

Aside from that for the argument of upgrading now I wouldn't expect too many issues from the typescript upgrade itself and generally typescript upgrades are pretty safe (and any issues are usually just because the older typescript version allowed something that was type unsafe that you should fix anyway) but bring a lot more expressivity and I wouldn't hold back based on that

Reply:
https://github.com/balena-io-modules/balena-semver/pull/81